web/common.inc.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+
+#### Build utility methods + definitions ####
+
+compiler="npm run tsc --"
+compilecmd="$compiler"
+
+# Compiles all build products corresponding to the specified target.
+# This should be called from the working directory of a child project's
+# build script.
+#
+# ### Parameters
+#
+# * 1: `product`    the product's source path under src/
+#
+# ### Example
+#
+# ```bash
+#   compile engine/main
+# ```
+compile ( ) {
+  if [ $# -lt 1 ]; then
+    builder_die "Scripting error: insufficient argument count!"
+  fi
+
+  local COMPILE_TARGET=$1
+
+  $compilecmd -b src/$COMPILE_TARGET -v
```

Comment:
```suggestion
  local COMPILE_TARGET="$1"

  tsc -b "src/$COMPILE_TARGET" -v
```

Reply:
Keep in mind that `$1` should correspond to a `:target` - there's something wrong if the parameter contains whitespace.

Reply:
I put in quote marks for paths as a matter of habit because it means less reasoning about them and accidentally missing an important one.